0004/README.md

```diff
@@ -0,0 +1,524 @@
+
+## Summary
+
+This document proposes an architecture for policy evaluation over SLSA
+provenance, for software packages that are managed with npm, the package manager
+for Node.js. When npm installs a software package and its transitive
+dependencies, it records the installed packages in a lock file
+(package-lock.json). The information that is written to these files is a primary
+input for a proposed SLSA provenance policy evaluator.
+
+With few differences, the approach presented here is expected to generalize to
+other lock file-based software environments such as Go (go.mod and go.sum), PHP
+(composer.lock), Python (Pipfile.lock), and Ruby (Gemfile.lock).
+
+## Terminology
+
+*   A *provenance policy* associates a package name with expectations for the provenance attestation properties for a collection of package versions. 
+
+*   A package version *satisfies* a provenance policy if its provenance attestation properties meet the policy expectations for the package name.
+
+## Objectives
+
+Evaluate policies over SLSA provenance for npm software packages.
+
+Non-objectives:
+
+-  Policies for what direct or indirect package dependencies (or dependency
+    versions) to expect for an npm software package; policies for vulnerability
+    scans.
+
+## Overview
+
+Here, we highlight the main architecture elements, their roles, and limitations.
+We present more detail in subsequent sections, with additional detail in the
+Appendix.
+
+### Extended software supply chain model
+
+![Extended model](extended-model.svg)
+
+The figure shows a generic supply chain for software packages, from source repository
+to installation, extended with two SLSA provenance policy evaluation points:
+
+-  When a software package is published to a package registry.
+-  When a software package is installed from a package registry.
+
+A SLSA provenance policy contains expectations for provenance, as discussed in later
+sections.
+
+SLSA Provenance expectations may be evaluated at other points inside or outside
+the above pipeline. Examples are a pre-build provenance check for build inputs,
+or an organization-wide proxy that checks SLSA provenance for package downloads.
+SLSA Provenance expectations may also differ between development versus
+production environments, or private versus public registries. We will not cover
+these differences in detail, but expect that the use cases can use similar SLSA
+provenance expectations as described here.
+
+### SLSA provenance expectations, benefits and limitations
+
+Each SLSA provenance policy covers multiple versions for a specific package
+name. For example, the policy for package ‘foo' could expect that each covered
+version:
+
+-  Has SLSA provenance that is signed with a key that is bound to a trusted
+    identity. The key could be long-lived, rotated regularly, or ephemeral, and
+    the identity could be a builder, individual, or organization.
+-  The provenance contains claims that:
```

Comment:
What are these claims? Do they come from one or more attestations? If more, are they different types of attestations beyond just (build) provenance?

Reply:
This text considers only *provenance* attestations for a specific package version artifact. If I suggest otherwise then I should fix that.

There may be information to limit  trust  based on builder ID or other context.  Those details are not covered here.